platform/server-api/src/main/java/org/motechproject/server/api/PomInformation.java

```diff
@@ -0,0 +1,162 @@
+package org.motechproject.server.api;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Parent;
+import org.apache.maven.model.Repository;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonatype.aether.graph.Dependency;
+import org.sonatype.aether.repository.RemoteRepository;
+import org.sonatype.aether.util.artifact.DefaultArtifact;
+import org.sonatype.aether.util.artifact.JavaScopes;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Holds all important information about POM.
+ */
+public class PomInformation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PomInformation.class);
+
+    private PomInformation parentPomInformation;
+
+    private Parent parent;
+    private List<Dependency> dependencies;
+    private Properties properties;
+    private List<RemoteRepository> repositories;
+
+    public void parsePom(File file) {
+        try {
+            parsePom(new FileInputStream(file));
+        } catch (FileNotFoundException ex) {
+            LOGGER.error("Error while opening POM file", ex);
+        }
+    }
+
+    public void parsePom(InputStream inputStream) {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+
+            Model model = reader.read(inputStream);
+
+            if (dependencies == null) {
+                dependencies = new LinkedList<>();
+            }
+            if (repositories == null) {
+                repositories = new LinkedList<>();
+            }
+
+            parent = model.getParent();
+            setPropertiesFromModel(this, model);
+
+            for (org.apache.maven.model.Dependency dependency : model.getDependencies()) {
+                if (!"test".equalsIgnoreCase(dependency.getScope())) {
+                    dependencies.add(new Dependency(new DefaultArtifact(
+                            (dependency.getGroupId().contains("${")) ? parent.getGroupId() : dependency.getGroupId(),
```

Comment:
This is a bit of a stretch, why not just parse it the same way we do with versions?